packages/component-utilities/src/formatting.js

```diff
@@ -199,3 +228,16 @@ function maybeExtractFormatTag(columnName) {
 		return undefined;
 	}
 }
+
+/**
+ * Formats a value to whichever format is passed in
+ * @param {*} value the value to be formatted
+ * @param {string} format string containing an Excel-style format code, or a format name matching a built-in or custom format
+ * @returns a formatted value
+ */
+export function fmt(value, format) {
+	let formatObj = getFormatObjectFromString(format);
+	let valueType = inferValueType(value);
+	formatObj.valueType = valueType;
```

Comment:
Looks like getFormatObjectFromString handles this already, i.e `getFormatObjectFromString(format, inferValueType(value));`

Reply:
I think this was introduced from a change on main, this wasn't part of this PR